### DISCUSSION TITLE
Once a week is not every 7 days

### DISCUSSION THREAD

**Discussion by StasChour** [iSoron/uhabits#1851] [Help & FAQ]
Hi All,

I tried every way possible but I want to set a habit for once a week (meaning it must be performed once per week no matter the day).
When I try to set it for once per week, the application forces me to once every 7 days, which is not the same!

for example, in week 1 I performed a habit on Sunday, and in week 2 I performed a habit on Thursday. For me it's once a week regardless.
However, the application forces me to once every 7 days no matter what which breaks the streak and I must fill the entries with skips...

more than 1 time a week works fine, so why not enable it once a week? Again once a week does not equal every 7 days...

Is there a solution to this?

**Comment by turkerozturk** [comment]
From FAQ page;

"How can I specify that a habit should be performed on certain days of the week?

You don't need to specify this. The app will figure it out by itself. You just need to specify how many times would you like to repeat the habit in a certain interval of time. For example, if you want to repeat a certain habit every Monday and Friday, simply specify that you want to repeat the habit 2 times per week, and then start practicing your habit. If, during your first week, you indeed perform your habit on Monday and Friday, the app will automatically add a checkmark on the following Saturday and Sunday. As long as you keep your routine, the checkmarks should appear automatically on the correct days."

**Reply by goskamich** [reply]
This tip from the FAQ does not address the above problem, it just shows that the application forces you to perform the habit on the same day every week in which you performed it the first time.  And the user (and me too) wants to be able to freely decide on which day of the week to perform the task, without breaking the streak.  when you set it automatically once a week, the condition jumps the field higher and is checked every 7 days, and this is not the same :(

**Reply by turkerozturk** [reply]
What if you select dash option "-" instead of check or uncheck? With it, you will postpone that day from calculation? I understood your requirement and sorry for my unrelated solution offer. I examined its SQLite database, each the "Habits" table has "freq_den" and "freq_num" fields. For example, if I select 2 times a month from user interface, then "freq_den" = 30 and "freq_num" = 2. These are all about timing. It is not based on Calendar, relative to first occurance then. For me, current software is very useful. Maybe in the future, I can have the same requirements like you I dont know. Then I can technically understand you. Anyway this software is a miracle to me. Until now, I didnt aware of habit tracker software type. Calendars and too lists are not filling the gap. Please trythe  dash option "-" instead of check or uncheck to postpone, maybe it can be a workaround.

**Reply by goskamich** [reply]
with "-" works, but I think it's a matter of adding the code to the database.  because if you say you want to do something 2 times/3 times etc. a week/month, it works and counts well.  In the application you also mark which day is the beginning of your week, so this can probably be done as the next update :) Apart from this one problem, I also think that the application is wonderful

**Comment by goskamich** [comment]
This also irritates me, because if you set it once a week, it automatically changes every 7 days.  and it is not the same.  I want to complete the task any time during the week, it doesn't have to be on Friday, but in these settings the application suggests every Friday.  completing the task on a different day breaks the streak.  does anyone have a solution for this?

**Comment by JenniferFalkenstein** [comment]
Same here, I want to do a task 3 times per week, no matter which day... hopefully this will get implemented some day
